packages/@react-aria/combobox/test/useComboBox.test.js

```diff
@@ -171,4 +172,55 @@ describe('useComboBox', function () {
     expect(toggleSpy).toHaveBeenCalledTimes(0);
     expect(buttonProps.isDisabled).toBeTruthy();
   });
+
+  it('should hide elements when opened', () => {
+    jest.spyOn(overlays, 'ariaHideOutside');
+
+    let {result: state} = renderHook((props) => useComboBoxState(props), {
+      initialProps: {...props, allowsEmptyCollection: true}
+    });
+
+    act(() => {
+      state.current.open();
+    });
+
+    renderHook((props) => useComboBox(props, state.current), {
+      initialProps: props
+    });
+
+    expect(overlays.ariaHideOutside).toHaveBeenCalledTimes(1);
+    expect(overlays.ariaHideOutside).toHaveBeenCalledWith([
+      props.inputRef.current,
+      props.popoverRef.current
+    ]);
+  });
+
+  it('should hide elements only when popover element available', () => {
+    jest.spyOn(overlays, 'ariaHideOutside');
+
+    let {result: state} = renderHook((props) => useComboBoxState(props), {
+      initialProps: {...props, allowsEmptyCollection: true}
+    });
+
+    act(() => {
+      state.current.open();
+    });
+
+    let {rerender} = renderHook(
+      (props) => useComboBox(props, state.current),
+      {
+        initialProps: {...props, popoverRef: {current: null}}
```

Comment:
Sorry for the delay on reviewing this. I finally came back and I noticed why this test is passing when it shouldn't be.

this line here is creating a popoverRef object which is different than the one in the `props` object. This means that when we run the rerender, the effect will trigger because popoverRef has "changed"

a more correct way to test it would be
```
    props.popoverRef.current = null;

    let {rerender} = renderHook(
      (props) => useComboBox(props, state.current),
      {
        initialProps: props
      }
    );

    expect(overlays.ariaHideOutside).toHaveBeenCalledTimes(0);

    props.popoverRef.current = document.createElement('div');

    // rerender with popover element available
    rerender(props);
```

which I believe would fail.
Unfortunately, it looks like we changed to swc/jest and it doesn't like mocking like this. I'm not sure why.

It would probably be easier to create the test at the React Aria Component level instead of testing at the hook level.

Reply:
It'd be good to figure out why the mui popover's ref isn't available yet as I'm not sure what else we could/would trigger another rerender on